app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt

```diff
@@ -97,13 +120,14 @@ fun AssistantScreen(tripsViewModel: TripsViewModel, navigationActions: Navigatio
                           keyboardController?.hide()
                           if (uiState !is UiState.Loading) {
                             tripsViewModel.sendActivitiesPrompt(
-                                trip, prompt, provideFinalActivities)
+                                trip, prompt, interests, provideFinalActivities, useInterests)
```

Comment:
I have a small suggestion: 
Even when useInterests is false, we're still collecting and passing the interests list, though it won't be used. We could optimize this by moving the interests collection inside the sendActivitiesPrompt call, something like:

tripsViewModel.sendActivitiesPrompt(
    trip, 
    prompt, 
    if (useInterests) userViewModel.user.collectAsState().value?.interests ?: emptyList() else emptyList(),
    provideFinalActivities, 
    useInterests
)

Reply:
Thanks for the suggestion! Indeed, passing the interests as argument when `useInterests` was false was redundant. Instead I removed `useInterests` from the parameters, but if it's false I pass an empty list as a parameter instead of the user's interests.